common/src/main/java/com/hortonworks/registries/common/ha/LeadershipClient.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hortonworks.registries.common.ha;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * Client to connect to target systems(like ZooKeeper) which give a way to elect a leader among the participants.
+ */
+public interface LeadershipClient {
```

Comment:
LeaderElection? LeaderElectionClient? LeaderElector? Leadership is not a name very commonly used